internal/commands/login/inputs_test.go

```diff
@@ -229,27 +239,39 @@ func TestLoginInputs(t *testing.T) {
 		})
 	}
 
-	t.Run("should output error message if browser cannot be opened", func(t *testing.T) {
+	t.Run("should output error message and prompt for credentials if browser cannot be opened", func(t *testing.T) {
 		openBrowserFunc := func(url string) error {
 			return errors.New("there was an issue opening your browser")
 		}
 
 		out := new(bytes.Buffer)
-		ui := mock.NewUIWithOptions(
+		console, _, ui, consoleErr := mock.NewVT10XConsoleWithOptions(
 			mock.UIOptions{OpenBrowserFn: openBrowserFunc},
 			out,
 		)
 
-		i := inputs{
-			AuthType: authTypeLocal,
-			Username: "username",
-			Password: "password",
-		}
+		assert.Nil(t, consoleErr)
+		defer console.Close()
+
+		doneCh := make(chan (struct{}))
+		go func() {
+			defer close(doneCh)
+			console.ExpectString("Username")
+			console.SendLine("username")
+			console.ExpectString("Password")
+			console.SendLine("password")
+			console.ExpectEOF()
+		}()
+
+		i := inputs{AuthType: authTypeLocal}
 
 		profile := mock.NewProfile(t)
 		assert.Nil(t, i.Resolve(profile, ui))
 
-		assert.Equal(t, "there was an issue opening your browser\n", out.String())
+		console.Tty().Close() // flush the writers
+		<-doneCh              // wait for procedure to complete
+
+		assert.True(t, strings.HasPrefix(out.String(), "there was an issue opening your browser"), "should show error")
```

Comment:
out of curiosity, what's the whole `out.String()` output?  wondering if we can do better/stricter than just a `HasPrefix` check

Reply:
the mock console prompts for the username/password
`"there was an issue opening your browser\r\n\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0m\x1b[?25l\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0mu\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0mus\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0muse\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0muser\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0musern\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0muserna\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0musernam\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0musername\x1b[?25h\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0m\x1b[36musername\x1b[0m\r\n\x1b[1;92m? \x1b[0m\x1b[1;99mPassword \x1b[0m\x1b[?25l\x1b7\x1b[999;999f\x1b[6n\x1b8\x1b[?25h\x1b[6n********\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1E"`

Reply:
gross